### PR TITLE
add job-image to github-actions, update to use github-actions only secrets

### PR DESCRIPTION
**Pull request by anodelman**
## Description
Another day, another test.  Plus, switching over to github-actions only secrets to mitigate risk.

## Related Issues
List related issues.

## Testing
Tested green on github.

## Checklist

 - [ ] I made sure to update `CHANGELOG.md`.
   
   Remember, the CHANGELOG needs to mention:
    + Any new features
    + Any changes to our included version of Envoy
    + Any non-backward-compatible changes
    + Any deprecations
 
 - [X] This is unlikely to impact how Ambassador performs at scale.
 
   Remember, things that might have an impact at scale include:
    + Any significant changes in memory use that might require adjusting the memory limits
    + Any significant changes in CPU use that might require adjusting the CPU limits
    + Anything that might change how many replicas users should use
    + Changes that impact data-plane latency/scalability
 
 - [X] My change is adequately tested.
 
   Remember when considering testing:
    + LEGACY MODE TESTS DO NOT RUN FOR EVERY PR. If your change is affected by legacy mode, you need
      to run legacy-mode tests manually (set `AMBASSADOR_LEGACY_MODE=true` and run the tests).
      (This will be fixed soon.)
    + Your change needs to be specifically covered by tests.
       + Tests need to cover all the states where your change is relevant: for example, if you add a behavior that can be enabled or disabled, you'll need tests that cover the enabled case and tests that cover the disabled case. It's not sufficient just to test with the behavior enabled.
    + You also need to make sure that the _entire area being changed_ has adequate test coverage.
       + If existing tests don't actually cover the entire area being changed, add tests.
       + This applies even for aspects of the area that you're not changing – check the test coverage, and improve it if needed!
    + We should lean on the bulk of code being covered by unit tests, but...
    + ... an end-to-end test should cover the integration points
 
 - [ ] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.

### CODE REVIEW

File: .github/workflows/execute-pytests.yml [hunk 1]
```diff
@@ -83,8 +83,8 @@ jobs:
         if: ${{ matrix.test != 'unit' }}
         uses: docker/login-action@v1
         with:
-          username: ${{ secrets.DOCKER_BUILD_USERNAME }}
-          password: ${{ secrets.DOCKER_BUILD_PASSWORD }}
+          username: ${{ secrets.GH_DOCKER_BUILD_USERNAME }}
+          password: ${{ secrets.GH_DOCKER_BUILD_TOKEN }}
```
Comment (acookin) [inline .github/workflows/execute-pytests.yml:87]: looks good assuming these are in keybase and are easy to find (which im sure you posted about in slack but honestly i can't remember)

File: .github/workflows/job-image.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+jobs:
+  execute-generate:
+    runs-on: ubuntu-latest
+    name: job-image
+    env:
+      AWS_ACCESS_KEY_ID: ${{ secrets.GH_AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.GH_AWS_SECRET_ACCESS_KEY }}
+      AWS_EC2_METADATA_DISABLED: true
+      DEV_REGISTRY: ${{ secrets.DEV_REGISTRY }}
```
Comment (acookin) [inline .github/workflows/job-image.yml:30]: non blocker but is there a way to do env variables set in github that aren't secrets? cuz we don't actually care about exposing this value, but we want to set it at the github repo level if that makes sense (circle didn't have a way to do this)

**Review by acookin — APPROVED** [review]
looks good just some genuine questions about things in github actions :)

File: .github/workflows/execute-pytests.yml (hunk 1 above)
Comment (anodelman) [reply .github/workflows/execute-pytests.yml:87]: Yup, totally in keybase.

File: .github/workflows/job-image.yml (hunk 2 above)
Comment (anodelman) [reply .github/workflows/job-image.yml:30]: I'll look things over - but I think that there are only secrets same as CircleCI.

**Pull request merged**